packages/common/src/utils/challenges.ts

```diff
@@ -350,41 +350,57 @@ export const makeOptimisticChallengeSortComparator = (
     ChallengeName.PlayCount10000
   ]
 
-  const getSortKey = (challenge: OptimisticUserChallenge) => {
-    if (challenge.claimableAmount > 0) {
+  return (id1, id2) => {
+    const userChallenge1 = userChallenges[id1]
+    const userChallenge2 = userChallenges[id2]
+
+    if (!userChallenge1 || !userChallenge2) {
       return 0
     }
+
+    // Priority 1: Claimable challenges come first
     if (
-      isNewChallenge(challenge.challenge_id) &&
-      challenge.state !== 'disbursed'
+      userChallenge1.claimableAmount > 0 &&
+      userChallenge2.claimableAmount <= 0
     ) {
-      return 1
+      return -1
     }
-    if (challenge.state === 'disbursed') {
-      return 2
+    if (
+      userChallenge2.claimableAmount > 0 &&
+      userChallenge1.claimableAmount <= 0
+    ) {
+      return 1
     }
-    return 3
-  }
 
-  return (id1, id2) => {
-    const userChallenge1 = userChallenges[id1]
-    const userChallenge2 = userChallenges[id2]
+    // Priority 2: New and not disbursed challenges come next
+    const isNewAndNotDisbursed = (userChallenge: OptimisticUserChallenge) =>
+      isNewChallenge(userChallenge.challenge_id) &&
+      userChallenge.state !== 'disbursed'
```

Comment:
this might get weird with aggregate challenges. why not use the claimable amount logic like above?

Reply:
oh bc those should already be handled. nvm